Mathlib/Topology/Algebra/InfiniteSum/Order.lean

```diff
@@ -346,6 +346,13 @@ theorem multipliable_mabs_iff [CommGroup α] [LinearOrder α] [IsOrderedMonoid 
 
 alias ⟨Summable.of_abs, Summable.abs⟩ := summable_abs_iff
 
+-- This should also work for any `AddCommGroup α` with `IsOrderedCancelAddMonoid α`:
+-- we are only use multiplication by `-1` here.
+theorem Summable.alternating [Ring α] [LinearOrder α] [IsOrderedRing α]
+  [UniformSpace α] [IsUniformAddGroup α] [CompleteSpace α] {f : ℕ → α} (hf : Summable f) :
+    Summable (fun n => (-1) ^ n * f n) :=
+  Summable.of_abs (by simpa [summable_abs_iff])
```

Comment:
```suggestion
theorem Summable.alternating [Ring α]
    [UniformSpace α] [IsUniformAddGroup α] [CompleteSpace α] {f : ℕ → α} (hf : Summable f) :
    Summable (fun n => (-1) ^ n * f n) := by
  apply Summable.even_add_odd
  · simp only [even_two, Even.mul_right, Even.neg_pow, one_pow, one_mul]
    exact hf.comp_injective (mul_right_injective₀ (two_ne_zero' ℕ))
  · simp only [pow_add, even_two, Even.mul_right, Even.neg_pow, one_pow, pow_one, mul_neg, mul_one,
      neg_mul, one_mul]
    apply Summable.neg
    apply hf.comp_injective
    exact (add_left_injective 1).comp (mul_right_injective₀ (two_ne_zero' ℕ))
```
no need for the linear order and the ordered ring assumption here. With that, you can also weaken the assumptions in `Summable.tendsto_alternating_series_tsum`.